File: Angular/fileshare/src/app/classes/report.ts
import {User} from './user';

export class Report {

  private _id: number;
  private _reported: User;
  private _description: string;
  private _date: Date;

  public constructor(id: number, reported: User, description: string, date: Date) {
    this._id = id;
    this._reported = reported;
    this._description = description;
    this._date = date;
  }

  public get id(): number {
    return this._id;
  }

  public get reported(): User {
    return this._reported;
  }

  public get description(): string {
    return this._description;
  }

  public get date(): Date {
    return this._date;
  }
}

File: fileshare/src/main/java/alkfejl/bead/fileshare/model/Report.java
package alkfejl.bead.fileshare.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "REPORT")
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Report extends BaseEntity {

    /*@JoinColumn
    @ManyToOne(targetEntity = User.class)
    private User reporter;
*/
    @JoinColumn
    @ManyToOne(targetEntity = User.class)
    private User reported;

    @Column(nullable = false)
    private String description;

    @Column
    private Date date = new Date();
}